### PR TITLE
Added golangci.yml file to repo

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

This PR adds the golangci.yml file and fixes the pending issues remaining for it to pass.

### CODE REVIEW

**Comment by agnivade** [comment]
Waiting on the mattermost/plugin-ci orb to be updated to include golangci binary.

**Comment by hanzei** [comment]
@agnivade What do you think about submitting a PR to https://github.com/mattermost/mattermost-plugin-starter-template first? Then we could unify the config file used in all plugin repos.

**Comment by agnivade** [comment]
Sure, that's fine. But IMO, I don't think it needs to block this from going in.

**Comment by agnivade** [comment]
golangci-lint is still not installed. @hanzei - does this depend on a new orb being released or something ?

**Comment by hanzei** [comment]
@agnivade Yes, a new release of the orb is needed. https://mattermost.atlassian.net/browse/IS-316 is tracking this request.

File: Makefile [hunk 1]
```diff
@@ -55,7 +65,7 @@ endif
 .PHONY: govet
 govet:
 ifneq ($(HAS_SERVER),)
-	@echo Running govet
+	@echo Running govet. (This is deprecated. Use make golangci-lint)
```
Comment (hanzei) [inline Makefile]: Why not just remove it?

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - gofmt
+    - gosimple
+    - govet
+    - ineffassign
+    - structcheck
+    - unconvert
+    - unused
+    - varcheck
+    # TODO: enable this later
+    # - errcheck
```
Comment (hanzei) [inline .golangci.yml]: Especially for https://github.com/mattermost/mattermost-plugin-starter-template I would push for include even more linters. https://github.com/mattermost/mattermost-marketplace/blob/master/.golangci.yml might be a good source for inspiration.

But this request in not blocking the PR from going in.

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+service:
+  golangci-lint-version: 1.21.0
```
Comment (hanzei) [inline .golangci.yml]: Let's just update now
```suggestion
  golangci-lint-version: 1.23.0
```

File: Makefile (hunk 1 above)
Comment (agnivade) [reply Makefile]: For compatibility reasons ? If someone is used to running `make govet` or `make gofmt`, this gives them some time to migrate to new commands. We did the same for the main repo.

File: .golangci.yml (hunk 2 above)
Comment (agnivade) [reply .golangci.yml]: Sounds good.

File: Makefile (hunk 1 above)
Comment (hanzei) [reply Makefile]: I see the need for migration time on the main repo as there are >100 developers. But given that this is small repo, I would just do a clean cut. 1/5

Comment (agnivade) [reply Makefile]: Ok, I will also make an announcement in the ~toolkit channel once this gets merged.

Comment (hanzei) [reply Makefile]: Maybe even ~Developers.

**Comment by marianunez** [comment]
I currently don't have enough domain knowledge on golangci to do a proper review. Deferring my review to @cpanato.

**Comment by prapti** [comment]
@agnivade What in particular are we wanting to test in this PR? Is there a Jira ticket that can be linked in the PR description and can some test steps be added to it? Thank you!

**Comment by agnivade** [comment]
@prapti - Just some basic smoke tests are enough. I can create a JIRA ticket if you prefer, but there's nothing much to add there.

**Comment by agnivade** [comment]
@hanzei - Do you know what's the status of the orb being released ?

**Comment by hanzei** [comment]
@agnivade DevOps will take care of this. https://mattermost.atlassian.net/browse/IS-316 is tracking the process.

**Comment by cpanato** [comment]
replacing me with @metanerd since the build engineers are responsible for the pipelines

**Comment by prapti** [comment]
> @prapti - Just some basic smoke tests are enough. I can create a JIRA ticket if you prefer, but there's nothing much to add there.

@agnivade Probably won't need a ticket if it's mainly smoke testing the Github plugin. I'll give it a go. Thank you!

**Comment by prapti** [comment]
@agnivade @hanzei Do I need to setup anything other than a regular test server and a github plugin with the PR changes here for the smoke test (I'm not familiar with the golangci and mattermost/plugin-ci orb discussed above)?

**Comment by agnivade** [comment]
> Do I need to setup anything other than a regular test server and a github plugin with the PR changes here for the smoke test (I'm not familiar with the golangci and mattermost/plugin-ci orb discussed above)?

Nope, that's all that's needed.

**Comment by prapti** [comment]
@agnivade The plugin works normally. However, the plugin functions now appear on the top of the LHS (pls see screenshot). Is this by design? (Is it even happening because of the changes in this PR? If not, I can approve.)
Screen Shot 2020-01-27 at 8 18 54 PM

**Comment by agnivade** [comment]
I forgot how it looked for single team apps. Was it at the bottom ? Could you check with the latest master ?

**Comment by hanzei** [comment]
The behaviour should be:
- In a user is in only one team: Icons are shown on the top LHS
- In a user is in more then one team: Icons are shown on the bottom LHS

File: .golangci.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    - gofmt
+    - gosimple
+    - govet
+    - ineffassign
+    - structcheck
+    - unconvert
+    - unused
+    - varcheck
+    # TODO: enable this later
+    # - errcheck
```
Comment (metanerd) [inline .golangci.yml]: I appreciate the todo!

**Comment by agnivade** [comment]
On hold until the orb is released.

**Comment by hanzei** [comment]
@agnivade A new version of the orb is releases. I did merge `master` into this branch and fixed some follow up issues. Feel free to merge when you think it's ready.

**Comment by agnivade** [comment]
Awesome, thanks for pushing this.

**Pull request merged**